.github/workflows/platform-api-tests.yml

```diff
@@ -0,0 +1,36 @@
+name: platform-api-tests
+
+on:
+  #  pull_request:
+  #    branches: [ '*' ]
+  schedule:
+    - cron: '0 3 * * *'
+  workflow_dispatch:
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
+
+jobs:
+  platform-api-tests:
+    runs-on: macos-11
+    if: github.repository == 'apollographql/apollo-kotlin'
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c #v3.3.0
+
+      - name: Setup Java
+        uses: actions/setup-java@1df8dbefe2a8cbc99770194893dd902763bee34b #v3.9.0
+        with:
+          distribution: 'temurin'
+          java-version: 17
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@3fbe033aaae657f011f88f29be9e65ed26bd29ef #v2.3.3
+
+      - name: Run tests against the Apollo Platform API
+        env:
+          PLATFORM_API_TESTS_KEY: ${{ secrets.PLATFORM_API_TESTS_KEY }}
+        run: |
+          ./gradlew -p tests --no-daemon --console plain :platform-api:registerServiceApolloOperations
+          ./gradlew -p tests --no-daemon --console plain :platform-api:downloadServiceApolloSchemaFromRegistry
+          ./gradlew -p tests --no-daemon --console plain :platform-api:pushApolloSchema --key=$PLATFORM_API_TESTS_KEY --subgraph=subgraph1 --schema=platform-api/src/main/graphql/schema.graphqls --revision=1 --graphVariant=current
```

Comment:
Should we start adding `--no-daemon --console plain` in other places too?

Also nit: my favourite solution would be defining a task in `platform-api/build.gradle.kts`, `endToEndTest` or something like that so we have more logic in Kotlin files and less logic in yaml files.

Reply:
- `--console plain` is definitely useless, logs are already good in the other jobs that don't have it
- `--no-daemon` is probably only useful in jobs that have only one run, to save a few seconds. Here we have several, so it's probably actually slower for no good reason.

I'll remove

Reply:
[Pushed](https://github.com/apollographql/apollo-kotlin/pull/5483/commits/7ef9a43e2c2d29e1ab9b2ca4eb20963bfc0b417b) a tweak to move the logic to `build.gradle.kts`. Uncommented temporarily `pull_request: branches: [ '*' ]` to make sure this works.

Reply:
Added `--no-daemon` back since there's only one task again :)

Reply:
👍 It'll run so fast now ⚡